diff --git a/src/widget/wnumberpos.cpp b/src/widget/wnumberpos.cpp
--- a/src/widget/wnumberpos.cpp
+++ b/src/widget/wnumberpos.cpp
@@ -36,12 +36,9 @@ void WNumberPos::mousePressEvent(QMouseEvent* pEvent) {
         // Cycle through display modes
         if (m_displayMode == TrackTime::DisplayMode::ELAPSED) {
             m_displayMode = TrackTime::DisplayMode::REMAINING;
-        } else if (m_displayMode == TrackTime::DisplayMode::REMAINING) {
-            m_displayMode = TrackTime::DisplayMode::ELAPSED_AND_REMAINING;
-        } else if (m_displayMode == TrackTime::DisplayMode::ELAPSED_AND_REMAINING) {
+        } else {
             m_displayMode = TrackTime::DisplayMode::ELAPSED;
         }
-
         m_pShowTrackTimeRemaining->set(static_cast<double>(m_displayMode));
         slotSetTimeElapsed(m_dOldTimeElapsed);
     }
@@ -111,10 +107,10 @@ void WNumberPos::slotSetDisplayMode(double remain) {
     if (remain == 1.0) {
         m_displayMode = TrackTime::DisplayMode::REMAINING;
-    } else if (remain == 2.0) {
-        m_displayMode = TrackTime::DisplayMode::ELAPSED_AND_REMAINING;
     } else {
         m_displayMode = TrackTime::DisplayMode::ELAPSED;
+        if (remain != 0.0) {
+            m_pShowTrackTimeRemaining->set(0.0);
+        }
     }
-
     slotSetTimeElapsed(m_dOldTimeElapsed);
 }
diff --git a/src/skin/legacy/tooltips.cpp b/src/skin/legacy/tooltips.cpp
--- a/src/skin/legacy/tooltips.cpp
+++ b/src/skin/legacy/tooltips.cpp
@@ -869,8 +869,8 @@ void Tooltips::addDefaultTooltips() {
 
     add("track_time")
             << tr("Track Time")
-            << tr("Displays the elapsed and/or remaining time of the track loaded.")
-            << tr("Click to toggle between time elapsed/remaining time/both.")
+            << tr("Displays the elapsed or remaining time of the track loaded.")
+            << tr("Click to toggle between elapsed and remaining time.")
             << tr("Hint: Change the time format in Preferences -> Decks.");
 
     add("track_duration")
